oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/contexts.js

```diff
@@ -1,8 +1,47 @@
-import React from 'react'
+import React, { useMemo, useContext } from "react";
+import { useFormikContext } from "formik";
+import { useFormConfig } from "./hooks";
+import { ApiClient, DepositActions } from "../api";
+console.log(useFormConfig);
 
-export const FormConfigContext = React.createContext()
+export const FormConfigContext = React.createContext();
 
+export const FormConfigProvider = ({ children, value }) => {
+  return (
+    <FormConfigContext.Provider value={value}>
+      {children}
+    </FormConfigContext.Provider>
+  );
+};
 
-export const FormConfigProvider = ({children, value}) => {
-    return <FormConfigContext.Provider value={value}>{children}</FormConfigContext.Provider>
-}
+export const ApiClientContext = React.createContext();
+
+export const ApiClientProvider = ({
+  children,
+  clientClass = DepositActions,
+}) => {
+  const {
+    formConfig: { createUrl },
+  } = useFormConfig();
```

Comment:
We shouldnt be depending on access to form config here anymore, in order to provide api client to children. there might be some non-formik based apps needing to access the apis

Reply:
@mirekys but formConfig is the only place from where I can gain access to createUrl currently in our form apps. Also all the relevant data necessary for the form app to work are coming from it. Probably I could get input from HTML directly, too, but this would be just duplicating what formConfig is already providing us. Aren't all of our APPs, going to have the same principle of getting record from the HTML and passing it to javascript?